hooks/useNotification.tsx

```diff
@@ -0,0 +1,25 @@
+import { useRecoilState } from 'recoil'
+import { notificationState } from 'lib/recoil-store/atoms'
+import { INotification } from 'lib/interfaces'
+import { useEffect } from 'react'
+
+const useNotification = () => {
+    const [notification, setNotification] = useRecoilState<INotification>(notificationState)
+    const showNotification = (props: INotification) => {
+        setNotification({
+            ...notification,
+            ...props,
+        })
+    }
+    useEffect(() => {
+        if (notification?.open && notification?.timeout) {
```

Comment:
If we add the const `NOTIFICATION_TIMEOUT_NEVER` you should also consider it here, something like 
```ts
if (notificationData?.timeout && notificationData?.timeout !== NOTIFICATION_TIMEOUT_NEVER) {
...
```